sympy/functions/elementary/trigonometric.py

```diff
@@ -87,7 +87,7 @@ def _as_real_imag(self, deep=True, **hints):
     def _period(self, general_period, symbol=None):
         f = expand_mul(self.args[0])
         if symbol is None:
-            symbol = tuple(f.free_symbols)[0]
+            symbol = next(iter(f.free_symbols))
```

Comment:
What happens if there are no symbols?

Note that leaking StopIteration is bad because it gets caught by things like `map`, `filter` etc. It is much better to leak a different exception like IndexError.

I would generally only use `next` in the 2-arg form like `next(items, None)`.

Reply:
Yeah it raises a StopIterationError. We shouldn't be leaking the error here because of the invariant, right? This is only an issue for replaces where we expect KeyError (which should probably have custom handling to rethrow as something else like ValueError, AssertionError etc).

Reply:
Leaking StopIteration is bad because it can have unexpected effects on calling code that uses iterator consumers like `map`:
```python
In [4]: stuff = ['foo', 'bar', '', 'baz']

In [5]: first = lambda e: next(iter(e))

In [6]: first_letters = list(map(first, stuff))

In [7]: first_letters
Out[7]: ['f', 'b']
```
I would only ever use 1-arg next if I knew that I wanted to leak the StopIteration which is only ever true in the unusual case of the body of the `__next__` method of an iterator.

Reply:
If ruff has a fixer that replaces other code with `next(iter(obj))` then I think that is a dangerous fix and the fixer should be discouraged/removed.

Reply:
@oscarbenjamin It's marked as an `unsafe-fix` and requires the user to manually apply the suggested fix. It is not applied by default with pre-commit.

Reply:
Also, it is the only easy way to deterministically get the first element of a dict key, value, items() in O(1) time.